core.go

```diff
@@ -86,19 +86,32 @@ func (c Core) ListMultipartUploads(ctx context.Context, bucket, prefix, keyMarke
 	return c.listMultipartUploadsQuery(ctx, bucket, keyMarker, uploadIDMarker, prefix, delimiter, maxUploads)
 }
 
+// PartUploadReq contains inputs for PutObjectPart
+type PartUploadReq struct {
+	Bucket, Object, UploadID string
+	PartID                   int
+	Data                     io.Reader
+	Size                     int64
+	Md5Base64, Sha256Hex     string
+	SSE                      encrypt.ServerSide
+	CustomHeader, Trailer    http.Header
+}
+
 // PutObjectPart - Upload an object part.
-func (c Core) PutObjectPart(ctx context.Context, bucket, object, uploadID string, partID int, data io.Reader, size int64, md5Base64, sha256Hex string, sse encrypt.ServerSide) (ObjectPart, error) {
+func (c Core) PutObjectPart(ctx context.Context, r PartUploadReq) (ObjectPart, error) {
 	p := uploadPartParams{
-		bucketName:   bucket,
-		objectName:   object,
-		uploadID:     uploadID,
-		reader:       data,
-		partNumber:   partID,
-		md5Base64:    md5Base64,
-		sha256Hex:    sha256Hex,
-		size:         size,
-		sse:          sse,
+		bucketName:   r.Bucket,
+		objectName:   r.Object,
+		uploadID:     r.UploadID,
+		reader:       r.Data,
+		partNumber:   r.PartID,
+		md5Base64:    r.Md5Base64,
+		sha256Hex:    r.Sha256Hex,
+		size:         r.Size,
+		sse:          r.SSE,
 		streamSha256: true,
+		customHeader: r.CustomHeader,
```

Comment:
You can follow the pattern of `CompleteMultipartUpload()` @donatello have essential args as actual inputs

```
PutObjectPart(ctx context.Context, bucket, object, uploadID string, partID int, opts PutObjectPartOptions)
```

Because other than bucket, object, uploadID and partID rest of them are optional